packages/web-app-files/src/components/SideBar/Shares/Links/DetailsAndEdit.vue

```diff
@@ -0,0 +1,480 @@
+      return this.$gettextInterpolate(
+        this.$gettext('Expires in %{timeToExpiry} (%{expiryDate})'),
+        { timeToExpiry: this.expirationDateRelative, expiryDate: this.localExpirationDate },
+        true
+      )
+    },
+
+    viaTooltip() {
+      if (!this.link.indirect) {
+        return null
+      }
+      return this.$gettextInterpolate(
+        this.$gettext('Navigate to the parent (%{folderName})'),
+        { folderName: basename(this.link.path) },
+        true
+      )
+    },
+
+    passwordEnforcedForRole() {
+      const currentRole = this.availableRoleOptions.find((role) => {
+        return parseInt(this.link.permissions) === role.role.bitmask(false)
+      })
+
+      const canRead = currentRole.role.hasPermission(SharePermissions.read)
+      const canCreate = currentRole.role.hasPermission(SharePermissions.create)
+
+      if (this.passwordEnforced.read_only === true) {
+        return canRead && !canCreate
+      }
+      if (this.passwordEnforced.upload_only === true) {
+        return !canRead && canCreate
+      }
+      if (this.passwordEnforced.read_write === true) {
+        return canRead && canCreate
+      }
+      return false
+    },
+
+    passwortProtectionTooltip() {
+      return this.$gettext('This link is password-protected')
+    }
+  },
+  watch: {
+    newExpiration(expiration) {
+      this.updateLink({
+        link: {
+          ...this.link,
+          expiration
+        }
+      })
+    }
+  },
+
+  methods: {
+    ...mapActions(['createModal', 'hideModal', 'setModalInputErrorMessage']),
+
+    updateLink({
+      link = this.link,
+      dropRef = this.$refs.editPublicLinkDropdown,
+      onSuccess = () => {}
+    }) {
+      this.$emit('updateLink', { link, onSuccess })
+      dropRef.hide()
+    },
+    deleteLink() {
+      this.$emit('removePublicLink', { link: this.link })
+      this.$refs.editPublicLinkDropdown.hide()
+    },
+    showRenameModal() {
+      const modal = {
+        variation: 'passive',
+        title: this.$gettext('Edit name'),
+        cancelText: this.$gettext('Cancel'),
+        confirmText: this.$gettext('Save'),
+        hasInput: true,
+        inputValue: this.link.name,
+        inputLabel: this.$gettext('Link name'),
+        onCancel: this.hideModal,
+        onConfirm: (name) =>
+          this.updateLink({
+            link: {
+              ...this.link,
+              name
+            },
+            onSuccess: () => {
+              this.hideModal()
+            }
+          })
+      }
+
+      this.createModal(modal)
+    },
+
+    checkPasswordNotEmpty(password) {
+      if (password === '') {
+        return this.setModalInputErrorMessage(this.$gettext("Password can't be empty"))
+      }
+      return this.setModalInputErrorMessage(null)
+    },
+
+    showPasswordModal() {
+      const modal = {
+        variation: 'passive',
+        title: this.link.password ? this.$gettext('Edit password') : this.$gettext('Add password'),
```

Comment:
`this.$gettext(this.link.password ? 'Edit password' : 'Add password')`

Reply:
Don't think that's a viable way to trigger transifex though 🤔